### PR TITLE
Fix #4871 JobExecution entities should have a weak reference to MolgenisUser. [MANUAL MERGE]

### LINKED ISSUE #4871: JobExecution entities should have a weak reference to MolgenisUser
The ```JobExecution``` entities have an xref to the ```MolgenisUser``` entity. This should be a string. Change it for the following entities:

- ```AnnotationJobExecution```
- ```SortaJobExecution```
- ```FileIngestJobExecution```
- ```GavinJobExecution```

### PR DESCRIPTION
**Pull request by jjettenn**
#### Checklist
- [ ] Unit tests
- [ ] Updated testplan

Use the resisterd backends names to set the default backend.

### CODE REVIEW

File: molgenis-data-csv/src/main/java/org/molgenis/data/csv/CsvRepositoryCollection.java [hunk 1]
```diff
@@ -29,7 +30,7 @@
  */
 public class CsvRepositoryCollection extends FileRepositoryCollection
 {
-	public static final String NAME = "CSV";
+	public static final String NAME = RepositoryCollection.BACKEND.CSV.name();
```
Comment (fdlk) [inline molgenis-data-csv/src/main/java/org/molgenis/data/csv/CsvRepositoryCollection.java]: CSV doesn't really feel like a backend.
Plus isn't there possibly a one to many relation between RepositoryCollection class and actual backend instance of the RepositoryCollection?

**Comment by fdlk** [comment]
Tests failing

**Comment by jjettenn** [comment]
Test me please

**Comment by fdlk** [comment]
test this please

**Pull request merged**